diff --git a/sharedmem.c b/sharedmem.c
index c37a58d..c825d29 100644
--- a/sharedmem.c
+++ b/sharedmem.c
@@ -5,10 +5,10 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <inttypes.h>
-#ifndef ANDROID
+#ifdef RMTFS_USE_LIBUDEV
 #include <libudev.h>
 #else
-#include <sys/endian.h>
+#include <endian.h>
 #endif
 #include <stdbool.h>
 #include <stdio.h>
@@ -27,7 +27,7 @@ struct rmtfs_mem {
 	int fd;
 };
 
-#ifndef ANDROID
+#ifdef RMTFS_USE_LIBUDEV
 
 static int parse_hex_sysattr(struct udev_device *dev, const char *name,
 			     uint64_t *value)
